source/Mission.cpp

```diff
@@ -143,16 +144,34 @@ void Mission::Load(const DataNode &node)
 			description = child.Token(1);
 		else if(child.Token(0) == "blocked" && child.Size() >= 2)
 			blocked = child.Token(1);
-		else if(child.Token(0) == "deadline" && child.Size() >= 4)
-			deadline = Date(child.Value(1), child.Value(2), child.Value(3));
 		else if(child.Token(0) == "deadline")
 		{
-			if(child.Size() == 1)
-				deadlineMultiplier += 2;
-			if(child.Size() >= 2)
-				deadlineBase += child.Value(1);
-			if(child.Size() >= 3)
-				deadlineMultiplier += child.Value(2);
+			if(child.Size() >= 4)
+				deadline = Date(child.Value(1), child.Value(2), child.Value(3));
+			else
+			{
+				if(child.Size() == 1)
+					deadlineMultiplier += 2;
+				if(child.Size() >= 2)
+					deadlineBase += child.Value(1);
+				if(child.Size() >= 3)
+					deadlineMultiplier += child.Value(2);
+			}
+		}
+		else if(child.Token(0) == "route options")
+		{
+			if(child.HasChildren())
```

Comment:
"Lonely if": either move the `HasChildren()` check into the same condition as the token check, or add feedback for the "no children" branch.